### PR TITLE
add support for workspaces

### PR DESCRIPTION
**Pull request by abhinandan13jan**
# Addresses
https://issues.redhat.com/browse/ODC-5559

# Screenshots
workspace support
Screenshot from 2021-04-07 19-03-37
Screenshot from 2021-04-06 22-43-33
Screenshot from 2021-04-06 23-23-00

# Tests
Updated optional check for WorkspaceDefinitionList
Updated Section check for WorkspaceResourceLinkList
Updated test Data 

# Browser conformance
Chrome

### CODE REVIEW

**Comment by andrewballantyne** [comment]
FWIW, I added the optional workspaces here @abhinandan13jan: https://github.com/openshift/console/pull/8576

File: frontend/packages/pipelines-plugin/src/components/shared/workspaces/WorkspaceResourceLinkList.tsx [hunk 1]
```diff
@@ -29,72 +29,70 @@ const WorkspaceResourceLinkList: React.FC<WorkspaceResourceLinkListProps> = ({
...
+                  name={secret.secretName}
+                  kind={SecretModel.kind}
+                  namespace={namespace}
+                  displayName={displayName}
+                />
+              );
+            }
+            if (!workspace.volumeClaimTemplate) {
+              return <div key={workspace.name}>{workspace.name}</div>;
+            }
```
Comment (andrewballantyne) [inline frontend/packages/pipelines-plugin/src/components/shared/workspaces/WorkspaceResourceLinkList.tsx:89]: What scenario is this section for?

Comment (abhinandan13jan) [reply frontend/packages/pipelines-plugin/src/components/shared/workspaces/WorkspaceResourceLinkList.tsx:89]: This is listing the `emptyDir`s

Comment (andrewballantyne) [reply frontend/packages/pipelines-plugin/src/components/shared/workspaces/WorkspaceResourceLinkList.tsx:89]: Interesting... not sure I like the if condition or how it renders... personally I think it should be removed if you are not handling the case, but I'll fix it in a follow up PR.

Thanks for the info.

**Review by andrewballantyne — APPROVED** [review]

**Pull request merged**